src/main/java/org/sonar/ucfg/UCFGtoProtobuf.java

```diff
@@ -0,0 +1,151 @@
+      .addAllBasicBlocks(ucfg.basicBlocks().values().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()))
+      .build();
+    try (FileOutputStream fos = new FileOutputStream(filename)) {
+      protobufUCFG.writeTo(fos);
+    }
+
+  }
+
+  private static Ucfg.BasicBlock toProtobuf(BasicBlock basicBlock) {
+    Ucfg.BasicBlock.Builder builder = Ucfg.BasicBlock.newBuilder()
+      .setId(basicBlock.label().id())
+      .setLocation(toProtobuf(basicBlock.locationInFile()))
+      .addAllInstructions(basicBlock.calls().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()));
+
+    if (basicBlock.terminator().type() == Instruction.InstructionType.RET) {
+      Instruction.Ret ret = (Instruction.Ret) basicBlock.terminator();
+      builder.setRet(Ucfg.Return.newBuilder().setLocation(toProtobuf(ret.location())).setReturnedExpression(toProtobuf(ret.getReturnedExpression())));
+    } else {
+      Ucfg.Jump.Builder jump = Ucfg.Jump.newBuilder();
+      Instruction.Jump terminator = (Instruction.Jump) basicBlock.terminator();
+      jump.addAllDestinations(terminator.destinations().stream().map(Label::id).collect(Collectors.toList()));
+      builder.setJump(jump);
+    }
+    return builder.build();
+  }
+
+  private static Ucfg.Instruction toProtobuf(Instruction.AssignCall assignCall) {
+    return Ucfg.Instruction.newBuilder().setLocation(toProtobuf(assignCall.location()))
+      .setVariable(assignCall.getLhs().id())
+      .setMethodId(assignCall.getMethodId())
+      .addAllArgs(assignCall.getArgExpressions().stream().map(UCFGtoProtobuf::toProtobuf).collect(Collectors.toList()))
+      .build();
+  }
+
+  private static Ucfg.Expression toProtobuf(Expression expression) {
+    Ucfg.Expression.Builder builder = Ucfg.Expression.newBuilder();
+    if (expression.isConstant()) {
+      builder.setConst(Ucfg.Constant.newBuilder().setValue(((Expression.Constant) expression).value()).build());
+    } else {
+      builder.setVar(Ucfg.Variable.newBuilder().setName(((Expression.Variable) expression).id()).build());
+    }
+    return builder.build();
+  }
+
+  private static Ucfg.Location toProtobuf(@Nullable LocationInFile locationInFile) {
+    if (locationInFile == null) {
+      return Ucfg.Location.getDefaultInstance();
+    }
+    return Ucfg.Location.newBuilder()
+      .setFileId(locationInFile.getFileId())
+      .setStartLine(locationInFile.getStartLine())
+      .setStartLineOffset(locationInFile.getStartLineOffset())
+      .setEndLine(locationInFile.getEndLine())
+      .setEndLineOffset(locationInFile.getEndLineOffset())
+      .build();
+  }
+
+  public static UCFG fromProtobufFile(String filename) throws IOException {
+    try (FileInputStream fis = new FileInputStream(filename)) {
+      Ucfg.UCFG ucfg = Ucfg.UCFG.parseFrom(fis);
+      UCFGBuilder builder = UCFGBuilder.createUCFGForMethod(ucfg.getMethodId()).at(fromProtobuf(ucfg.getLocation()));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in method 'at' declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.ucfg%3Asonar-ucfg&pullRequest=1&issues=AWJOLOfQ2-ApEdl5p-w3&open=AWJOLOfQ2-ApEdl5p-w3)

[ISSUE_KEY]: AWJOLOfQ2-ApEdl5p-w3